flax/linen/experimental/layers_with_named_axes.py

```diff
@@ -30,10 +30,10 @@
 
 # Type annotations
 Array = jnp.ndarray
-Axes = Union[int, Iterable[int]]
-DType = jnp.dtype
+Axes = Union[int, Any]
```

Comment:
could we add a `IntSequence` (or something) to indicate what this `Any` is supposed to be for a human reader?

Reply:
Sounds like a great idea! Just as an FYI, there are many places in the code where pytype wanted something to be `Iterabale` and mypy wanted something to be `Sequence`. I couldn't make both happy so they ended up with `Any`

Reply:
that's both sad and funny... I wonder if there's a way to harmonize these two in our code better or to agitate for change - for now just another Any alias indicating we intend a sequence of ints should be ok